Update geopm_agent_frequency_map man page

cmcantalupo opened the pull request:
- Also changes behavior of the geopmagent command line interpreter to support strings
  for region hashes in the policy vector.
- Related to #1049

cmcantalupo committed:
Modify geopmagent CLI to hash strings found in policy vector.

- If a policy value cannot be parsed as a double, hash
  the string instead.
- This enables referencing region names as strings rather
  than the hashed value in support of the frequency map agent.

Change-Id: I3dbf348bc520d79d622a3b800da8ac226276ceb2
Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo committed:
Update the geopm_agent_frequency_map man page

- Related to #1049

Change-Id: I281d50eab1ad122cf648f4e2264380ef59c58551
Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

dianarg commented on ronn/geopm_agent_frequency_map.7.ronn:
```diff
@@ -32,77 +32,170 @@ geopm_agent_frequency_map(7) -- agent for running regions at user selected frequ
-* `Control Loop Gate`:
-  The agent gates the Controller's control loop to a cadence
-  of 5ms.
+The frequency map agent selects CPU frequency in response to changes
```
selects -> adjusts?

dianarg commented on ronn/geopm_agent_frequency_map.7.ronn:
```diff
@@ -32,77 +32,170 @@ geopm_agent_frequency_map(7) -- agent for running regions at user selected frequ
-  of 5ms.
+The frequency map agent selects CPU frequency in response to changes
+in the application region being executed on each CPU.  The agent
+policy provides a map from a GEOPM region hash to a CPU frequency, see
```
, --> ;

dianarg commented on ronn/geopm_agent_frequency_map.7.ronn:
```diff
@@ -32,77 +32,170 @@ geopm_agent_frequency_map(7) -- agent for running regions at user selected frequ
+
+Note: this agent can be used to control the application CPUs and
+optionally the uncore to run at fixed frequencies if no region
+frequency map is provided.
```
this is useful for experiments sweeping over a range of core and/or uncore frequency

dianarg commented on ronn/geopm_agent_frequency_map.7.ronn:
```diff
@@ -32,77 +32,170 @@ geopm_agent_frequency_map(7) -- agent for running regions at user selected frequ
+  `FREQ_DEFAULT`: The operating frequency set by the agent on CPUs
+                  executing the application while the application is
+                  not within a policy-mapped region.  The processor's
+                  sticker frequency is selected for this parameter if
```
this means that running with this agent with NAN policy only could have lower performance than the monitor.

dianarg commented on ronn/geopm_agent_frequency_map.7.ronn:
```diff
@@ -32,77 +32,170 @@ geopm_agent_frequency_map(7) -- agent for running regions at user selected frequ
+            provided in the geopm report, and can be generated as
+            described in **geopm_hash(3)**.  In binary code, the
+            policy values are stored the double precision
+            representation of the 32 bit integer generated by the hash
```
32-bit

dianarg commented on ronn/geopm_agent_frequency_map.7.ronn:
```diff
@@ -32,77 +32,170 @@ geopm_agent_frequency_map(7) -- agent for running regions at user selected frequ
+
+## CONTROL LOOP RATE
+The agent gates the control loop to sample at 2 milliseconds and rate
+limit frequency changes to 500 Hz.
```
limits the rate of frequency changes?  could these be in the same units?

cmcantalupo replied:
Maybe overkill, but these two numbers are the same...just inverted.

cmcantalupo commented on ronn/geopm_agent_frequency_map.7.ronn:
```diff
@@ -32,77 +32,170 @@ geopm_agent_frequency_map(7) -- agent for running regions at user selected frequ
+                      2 * $(geopmread FREQUENCY_STEP board 0)))
+    $ freq_uncore=$(geopmread MSR::UNCORE_RATIO_LIMIT:MAX_RATIO board 0)
+    $ geopmagent -a frequency_map -p$freq_default,$freq_uncore
+      {"FREQ_MIN": 3500000000, "FREQ_MAX": 2400000000}
```
yuck

dianarg replied:
fix policy names

dianarg reviewed (changes requested):
Seems pretty close; a few typos.  One feature that may be confusing is that NAN for core frequency means sticker, but NAN for uncore means operate within the normal range.  Does it make sense for NAN to mean the same thing for the core frequency?  Then we don't need to worry about whether to make max or sticker the default.

cmcantalupo committed:
Address review comments

- Get rid of default default frequency.  Throw error instead.
- Clean up text for CONTROL LOOP RATE.
- Specify units for policy values.
- Some typo fixes.

Change-Id: I5327e13a90413c2e3ea81e8e499b653aee059dfe
Signed-off-by: Christopher M. Cantalupo <christopher.m.cantalupo@intel.com>

cmcantalupo reviewed (commented):
Updated the branch and I think I've addressed all concerns.
